_ap/15/index.adoc

```diff
@@ -0,0 +1,58 @@
+---
+num: 15
+category: "Dynamic Scaling"
+title: "Dynamic Scaling of Data Plane clusters"
+status: "Draft"
+authors:
+  - "Manyanda Chitimbo"
+  - "Steven Hawkins"
+  - "Tom Bentley"
+  - "Peter Braun"
+  - "Ramesh Reddy"
+---
+
+
+== Context and Problem Statement
+
+In the context of both cost management, as well as having the ability to scale to a (virtually) infinite number of manage  instances, 
+our services must have the capability to dynamically scale its underlying infrastructure. 
+This implies both scaling up the infrastructure to accommodate more customers, as well as scaling down the infrastructure to manage costs.
+
+There are a number of things to consider in the context of dynamic scaling:
+. Scaling up/down the number of nodes on an OSD 
+. Provisioning a new OSD
+. Removing an OSD instance
+. Pre-warm nodes to ensure hot capacity for maintaining managed instance provisioning SLOs and SLAs
+
+== Proposed solution
+
+=== Autoscaling of nodes
+
+Cluster Autoscaler will be utilized to allow nodes to be created on demand and removed when they are no longer needed. 
+See https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler#introduction[cluster-autoscaler#introduction] for more information.
+
+=== Pre-warming of nodes using Pause pods
+
+In order to ensure that the managed instance provisioning latency is not impacted by the time it takes for a new node to be provisioned, 
+it's possible to reserve hot capacity by configuring overprovisioning: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#how-can-i-configure-overprovisioning-with-cluster-autoscaler[how to configure overprovisioning]
+
+
+=== Cluster Full
+
+==== Use node sizing formula
+
+For binpacked solutions, a node sizing formula can be used to determine if a cluster is full or not based on the current number of maximum 
+nodes configured and the current workload. The fleetshard will then propagate the status of the cluster via the managed CR (custom resource) for the control plane
```

Comment:
Should we have introduced the idea of a "maximum number of configured nodes" before this point?

Reply:
I meant more a sentence to clarify where the maximum sizes _comes from_. i.e. is it a requirement that's imposed upon us (e.g. comes with using the autoscaler itself) vs. something that we've imposed upon ourselves.

Reply:
It is both i.e a managed service needs to define scaling limits (e.g a service might say, for each an OSD scale up to 100 nodes, and leave the remaining room for the expansion of deployed instances). At the same time the max size is a required parameter when configuring the autoscaler.